source/connection-monitoring-and-pooling/tests/pool-checkout-error-closed.yml

```diff
@@ -15,14 +15,22 @@ events:
   - type: ConnectionPoolCreated
     address: 42
     options: 42
+  - type: ConnectionCheckOutStarted
+    address: 42
   - type: ConnectionCheckedOut
+    address: 42
     connectionId: 42
   - type: ConnectionCheckedIn
+    address: 42
     connectionId: 42
   - type: ConnectionPoolClosed
     address: 42
+  - type: ConnectionCheckOutStarted
+    address: 42
+  - type: ConnectionCheckOutFailed
+    address: 42
+    reason: poolClosed
```

Comment:
This behavior is outlined in the spec (`ConnectionCheckOutFailed(reason="poolClosed"`)  but was not previously tested.